File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

	@Reference
	private BrandService brandService;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){			
		return brandService.findAll();
	}
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return brandService.findPage(page, rows);
	}
	
	/**
	 * 增加
	 * @param brand
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand){
		try {
            System.out.println(brand);
            System.out.println(brand.getName()+brand.getFirstChar());
			brandService.add(brand);
			System.out.println(brand.getId()+brand.getName()+brand.getFirstChar());
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	/**
	 * @Description: 增加品牌返回值为map
	 * @Param:
	 * @return:
	 * @Author: lxy
	 * @Date: 2020/10/4
	 */
	/*@RequestMapping("/add")
	public Map<String,Object> add(@RequestBody TbBrand brand){
		Map<String,Object> map = new HashMap<>();
		try {
			brandService.add(brand);
			map.put("success",true);
			map.put("message","增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			map.put("success",false);
			map.put("message","增加失败");
		}
        return map;
	}*/
	
	/**
	 * 修改
	 * @param brand
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand brand){
		try {
			brandService.update(brand);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbBrand findOne(Long id){
		return brandService.findOne(id);		
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Long [] ids){
		try {
			brandService.delete(ids);
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param brand
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand brand, int page, int rows  ){
		return brandService.findPage(brand, page, rows);		
	}

	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList() {
		return brandService.selectOptionList();
	}


}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
 //控制层 
app.controller('brandController' ,function($scope,$controller   ,brandService){	
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		brandService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		brandService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		brandService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var serviceObject;//服务层对象  				
		if($scope.entity.id!=null){//如果有ID
			serviceObject=brandService.update( $scope.entity ); //修改  
		}else{
			serviceObject=brandService.add( $scope.entity  );//增加 
		}				
		serviceObject.success(
			function(response){
				alert(response);//返回值为[object,object]
                alert(JSON.stringify(response));//返回值为{"message":"增加成功","success":true}
				//是一个json对象的格式，json常见的数据格式第三种，多个无序的数据对象，
				/*
				* 	3：多个无序的数据：对象
                  	   用{}包含起来，其元素必须由key-value组成，key是一个字符串，value是任意类型数据，
                  	   Key与value之间用:号隔开，两个key-value之间使用，号隔开，通过object.key的形式访问数据。
                      (对象表示为键值对，数据由逗号分隔，花括号保存对象，方括号保存数组)
                 	对象
                 	例子：{“name”:”sunwukong”, ”age”:18}*/

				/*所以这里response.success也就是通过object.key的形式访问数据，也就是说response.success的值为true，
				*这个时候就代表我们增加品牌的操作成功完成，这个时候我们就需要重新刷新页面。否则也就是我们增加或者修改的操作
				*失败那么就会弹出这个错误信息。
				* */
				if(response.success){
					//重新查询,这个方法是在baseController.js文件里面
		        	$scope.reloadList();//重新查询
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框
		alert($scope.selectIds);
		brandService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		brandService.search(page,rows,$scope.searchEntity).success(
			function(response){
				alert(page);
				alert(rows);
				alert(JSON.stringify($scope.searchEntity));
				alert(JSON.stringify(response));
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
    
});	
